triple-store/triple-store-api/src/main/java/com/powsybl/triplestore/api/PropertyBag.java

```diff
@@ -85,7 +85,11 @@ public double asDouble(String property, double defaultValue) {
         if (!containsKey(property)) {
             return defaultValue;
         }
-        return Double.parseDouble(get(property));
+        try {
+            return Double.parseDouble(get(property));
+        } catch (NumberFormatException x) {
+            return Double.NaN;
```

Comment:
rather than `NaN`, shouldn't it return `defaultValue`? In any case, I think we should at least have write a warning in `LOG`.